src/module/token.ts

```diff
@@ -221,6 +221,35 @@ export class LancerToken extends Token {
     // Make a clone so the cache can't be mutated
     return this._spaces.spaces.map(p => ({ ...p }));
   }
+
+  /**
+   * Given range and a filtering callback, returns whether certain targets
+   * are within that range.
+   */
+  areTargetsInRange(
+    maxRange: number,
+    callback: (o: QuadtreeObject<LancerToken>, distance: number) => boolean
+  ): boolean {
+    maxRange -= 1;
+    // Rough bounding box in which the hexes/squares will be searched
+    // From testing it seems that it goes to one hex further than necessary in the corners
+    // perhaps narrow it for performance
+    const aabb = new PIXI.Rectangle(
```

Comment:
A small potential optimization - for hex and gridless scenes, maybe we could use a circle instead? I don't know how different the collision performance is with a rectangle vs. circle, but a circle should omit a few tokens in the corners of the rectangle that are too far away.